metadata/metadata.go

```diff
@@ -47,18 +52,30 @@ func FromContext(ctx context.Context) (Metadata, bool) {
 		return nil, ok
 	}
 
-	// capitalise all values
-	newMD := make(map[string]string)
+	newMD := make(Metadata)
+
 	for k, v := range md {
-		newMD[strings.Title(k)] = v
+		key, err := base32.StdEncoding.DecodeString(strings.ToUpper(k))
+		if err != nil {
+			// try to be compatible with older micro versions
+			newMD[strings.Title(k)] = v
```

Comment:
You're not continuing here so it will do this and the thing below

Reply:
inattentive today , thanks